src/cdrizzlemap.c

```diff
@@ -13,580 +13,1072 @@
+};
+
+/**
+ * Set-up scanner structure for a polygon.
+ *
+ * This function finds minimum and maximum y-coordinates of the polygon
+ * vertices and splits all edges int left and right edges based on their
+ * horizontal position relative to the line connecting the top and bottom
+ * vertices of the polygon. It also copies the bounding box parameters
+ * xmin, xmax, ymin, ymax from the driz_param_t structure.
+ *
+ * @param[in] struct polygon *p.
+ * @param[in] struct driz_param_t - drizzle parameters (bounding box is used).
+ * @param[out] struct scanner *s - scanner structure to be initialized.
+ * @return 0 if successful and 1 if input polygon has only 2 vertices or less.
+ *
+*/
+int
+init_scanner(struct polygon *p, struct driz_param_t* par, struct scanner *s) {
+    int k, i1, i2;
+    int min_right, min_left, max_right, max_left;
+    double min_y, max_y;
+
+    s->left = NULL;
+    s->right = NULL;
+    s->nleft = 0;
+    s->nright = 0;
+
+    if (p->npv < 3) {
+        // not a polygon
+        s->overlap_valid = 0;
+        return 1;
+    }
 
-          if (fb * fc > 0.0) {
-            c = b;
-            fc = fb;
-            if (side == -1) {
-                fa *= 0.5;
-            }
-            side = -1;
+    // find minimum/minima:
+    min_y = p->v[0].y;
+    min_left = 0;
+    for (k = 1; k < p->npv; k++) {
+        if (p->v[k].y < min_y) {
+            min_left = k;
+            min_y = p->v[k].y;
+        }
+    }
 
-          } else if (fa * fb > 0.0) {
-            a = b;
-            fa = fb;
-            if (side == +1) {
-                fc *= 0.5;
-            }
-            side = +1;
+    i1 = mod(min_left - 1, p->npv);
+    i2 = mod(min_left + 1, p->npv);
+    min_right = ( p->v[i1].y < p->v[i2].y ) ? i1 : i2;
+    if (p->v[min_right].y <= min_y * (1.0 + copysign(VERTEX_ATOL, min_y))) {
+        if (p->v[min_left].x > p->v[min_right].x) {
+            k = min_left;
+            min_left = min_right;
+            min_right = k;
+        }
+    } else {
+        min_right = min_left;
+    }
 
-          } else {
-            /* if the product is zero, we have converged */
-            break;
-          }
+    // find maximum/maxima:
+    max_y = p->v[0].y;
+    max_right = 0;
+    for (k = 1; k < p->npv; k++) {
+        if (p->v[k].y > max_y) {
+            max_right = k;
+            max_y = p->v[k].y;
         }
+    }
 
-        if (n > m) {
-          return 1;
+    i1 = mod(max_right - 1, p->npv);
+    i2 = mod(max_right + 1, p->npv);
+    max_left = ( p->v[i1].y > p->v[i2].y ) ? i1 : i2;
+    if (p->v[max_left].y >= max_y * (1.0 - copysign(VERTEX_ATOL, max_y))) {
+        if (p->v[max_left].x > p->v[max_right].x) {
+            k = max_left;
+            max_left = max_right;
+            max_right = k;
         }
+    } else {
+        max_left = max_right;
+    }
 
-        xybounds->point[ipoint][idim] = b;
+    // Left: start with minimum and move counter-clockwise:
```

Comment:
I think the handling of separating the left and right edges could use a little more explanation. For example, a naive reading of this comment seems backwards from what I would expect. In that if we are starting with the edge that is at the y minimum shouldn't we go clockwise instead of counterclockwise (and the same issue for the comment on the right). What is going on with this could be better explained.